Issue #17 incorporate math into visualize progress

kirons323 opened the pull request:
Visualize Progress now works, will display the values for each particular stat for both what the character has and what they are expected to have at that time.  There are also now dropdowns in the Compare Character, Update, and Visualize Progress menus so that the user cannot select a character that they haven't added.

kirons323 committed:
Slight modifications to calculations

I modified some of the function calls in the calculations and moved that
code out of the Calculations.py file into CharacterData but the Stat
enum is causing syntax problems I don't know how to solve.  I also tried
making character selections dropdowns in the compareCharUi.py file but
the list is horizontal and selects everything (currently commented out).

wagnet committed:
Fixed some issues with visualization not being able to grab a stat value

kirons323 committed:
Finished Math Modelling

Math Modelling is now fully incorporated into the CharacterData class,
so the Calculations.py file should be obsolete.

kirons323 committed:
Fixed Corner Cases

Added a fix for corner cases for odd classes (Lord, Manakete, Chameleon,
Ballistician, Thief).

kirons323 committed:
Character Dropdowns

Now has dropdowns for character selection so the user cannot enter a
character that they had not yet added

kirons323 committed:
Fixed Same Sum Bug

Fixed bug where characters with the same sum of stats would overwrite
each other in the Suggest Team function.

wagnet commented on CharacterData.py:
```diff
@@ -2,110 +2,159 @@
+		return self.__growth_rate_class[game_class].rates
+
+	def predict_state(self, level, game_class, stat, game_data):
+		"""Creates a state for the average stats for a character with a given
```
This says it returns a State but is returning a float

wagnet commented on CharacterData.py:
```diff
@@ -2,110 +2,159 @@
+		Args:
+			level : an int representing an in game level
+			game_class: a str describing an in game class
+			stat : a str representing an in game stat
```
stat should probably be a Stat element instead of a str

kirons323 committed:
Fixed comment on function, added error checks

I added error checks so that no matter what the user enters the program
should not break.  It won't tell the user anything is wrong but it won't
break anything.
